DesignSafe-CI/portal
# Task/DES-1902 - Connect endpoints for versioning and amends

## Overview: ##
This PR is meant to complete the Amend and Revision pipelines. The process of amending and versioning a project is mostly complete.

## PR Status: ##


## Related Jira tickets: ##

* [DES-1902](https://jira.tacc.utexas.edu/browse/DES-1902)

## Testing Steps: ##
**NOTE: File copying and directory creation is not currently testable from the Dev environment. This may be possible to test in Staging though. If you plan to submit any dev project for publication, revision, or amendments please DM me on Slack with the correct designsafe.env settings to use. I'm open to have some group testing on this**
1. Navigate into an "other" project that has files.
2. Go to the "Prepare to Publish" section.
 - From this section you should be able to see options for revise or amend if your "other" project is published. If it is not, you can still access these through the browser console by running the `goVersion()` and `goAmend()` functions. Once you navigate you should be able to test the various sections of the pipeline.

## UI Photos:

<img width="774" alt="Screen Shot 2021-03-28 at 8 23 58 PM" src="https://user-images.githubusercontent.com/29575979/112775850-adcb1000-9003-11eb-9bc5-5fb377c91f75.png">
<img width="947" alt="Screen Shot 2021-03-30 at 6 12 00 PM" src="https://user-images.githubusercontent.com/29575979/113068297-ac285600-9183-11eb-8746-c87051e9cb16.png">
<img width="929" alt="Screen Shot 2021-03-30 at 6 12 14 PM" src="https://user-images.githubusercontent.com/29575979/113068301-acc0ec80-9183-11eb-8cbb-e29ff60d8dc5.png">
<img width="927" alt="Screen Shot 2021-03-30 at 6 14 32 PM" src="https://user-images.githubusercontent.com/29575979/113068403-d9750400-9183-11eb-895a-564a07d6fa6d.png">
<img width="928" alt="Screen Shot 2021-03-30 at 6 12 51 PM" src="https://user-images.githubusercontent.com/29575979/113068303-ad598300-9183-11eb-9ab9-bf8fb3abebf7.png">
